cli/src/main/java/com/devonfw/tools/ide/environment/AbstractEnvironmentVariables.java

```diff
@@ -149,6 +149,24 @@ public AbstractEnvironmentVariables extend(Path propertiesFilePath, EnvironmentV
     return new EnvironmentVariablesPropertiesFile(this, type, propertiesFilePath, this.context);
   }
 
+  private EnvironmentVariablesType incrementType(EnvironmentVariablesType type, Object rootSrc) {
+
+    EnvironmentVariables current = this;
+    while (current.getType() != type) {
+      current = current.getParent();
+      if (current == null) {
+        this.context.warning("During resolving of rootSrc {} the type {} was not found. The variable might not "
+            + "be resolved correctly", rootSrc, type);
+        return null;
+      }
+    }
+    EnvironmentVariables parent = current.getParent();
+    if (parent == null) {
+      return null;
+    }
+    return parent.getType();
+  }
```

Comment:
I still do not get why you added this method.
Did you consider using the existing `getByType` method?
This seems to do almost the same thing and has a JavaDoc explaining what it does.
Then you could also call the `resolve` method on the returned `EnvironmentVariables` and also you would not need to extend the resolve method with the additional `EnvironmentVariablesType` parameter.
Does this make sense or am I missing something?
Does this make sense or

Reply:
I changed the method, please have a look.

Reply:
We are getting closer. However, IMHO `resolvedVars` is never changed. Hence, it is the same as `this`. Hence, it is a pointless argument that can be omitted making the code much better to read and understand.
I might be mistaken. This is a complex story and understanding the code in the browser is not trivial. We can also have a video-chat together to quickly clarify this. But we almost reached the finishing line for merging. Thanks for your great work solving this complex problem 👍

Reply:
> String replacement = ((AbstractEnvironmentVariables) next).resolve(next.getFlat(variableName), variableName, recursion, rootSrc, rootValue, resolvedVars);

OK. Not easy to see. Maybe then we need some doc comment for other to understand this...

Reply:
I added javadoc to explain how `resolvedVars` is used and why it is necessary.